bool is_left
bool is_center
bool is_right
int64 area